files/validator.py

```diff
@@ -0,0 +1,398 @@
+        """
+        ---
+        dr_sites_primary_url: http://xxx.xx.xx.xxx:8080/ovirt-engine/api
+        dr_sites_primary_username: admin@internal
+        dr_sites_primary_ca_file: /etc/pki/ovirt-engine/ca.pem
+
+        # Please fill in the following properties for the secondary site:
+        dr_sites_secondary_url: http://yyy.yy.yy.yyy:8080/ovirt-engine/api
+        dr_sites_secondary_username: admin@internal
+        dr_sites_secondary_ca_file: /etc/pki/ovirt-engine_secondary/ca.pem
+        """
+        self.primary_url = var_file.get('dr_sites_primary_url')
+        self.primary_user = var_file.get('dr_sites_primary_username')
+        self.primary_ca = var_file.get('dr_sites_primary_ca_file')
+        # TODO
+        self.primary_password = "12"
+        self.secondary_url = var_file.get('dr_sites_secondary_url')
+        self.secondary_user = var_file.get('dr_sites_secondary_username')
+        self.secondary_ca = var_file.get('dr_sites_secondary_ca_file')
+        # TODO
+        self.secondary_password = "12"
+        self.prefix = prefix
+
+    def validate_primary(self):
+        isValid = True
+        if self.primary_url is None:
+            print(self.error_msg % (self.prefix, "url", "primary"))
+            isValid = False
+        if self.primary_user is None:
+            print(self.error_msg % (self.prefix, "username", "primary"))
+            isValid = False
+        if self.primary_password is None:
+            print(self.error_msg % (self.prefix, "password", "primary"))
+            isValid = False
+        if self.primary_ca is None:
+            print(self.error_msg % (self.prefix, "ca", "primary"))
+            isValid = False
+        return isValid
+
+    def validate_secondary(self):
+        isValid = True
+        if self.secondary_url is None:
+            print(self.error_msg % (self.prefix, "url", "secondary"))
+            isValid = False
+        if self.secondary_user is None:
+            print(self.error_msg % (self.prefix, "username", "secondary"))
+            isValid = False
+        if self.secondary_password is None:
+            print(self.error_msg % (self.prefix, "password", "secondary"))
+            isValid = False
+        if self.secondary_ca is None:
+            print(self.error_msg % (self.prefix, "ca", "secondary"))
+            isValid = False
+        return isValid
+
+    def connect_primary(self):
+        return self._connect_sdk(self.primary_url,
+                                 self.primary_user,
+                                 self.primary_password,
+                                 self.primary_ca)
+
+    def connect_secondary(self):
+        return self._connect_sdk(self.secondary_url,
+                                 self.secondary_user,
+                                 self.secondary_password,
+                                 self.secondary_ca)
+
+    def _connect_sdk(self, url, username, password, ca):
+        connection = sdk.Connection(
```

Comment:
You should somewhere close the connection.

Reply:
done